memory/model/nearest_neighbors/neighbor_gen.py

```diff
@@ -0,0 +1,39 @@
+from nearpy import Engine
```

Comment:
For imports, I tend to follow the format of:
1) Libraries included in the standard Python installation such as `os`, `argparse`, etc.
3) <Space>
2) Third-party libraries such as `numpy` ,`tensorflow`, etc.
3) <Space>
4) External AUTOLab libraries such as `autolab_core`
5) <Space>
6) Internal imports from inside this library.

This is just to keep things organized. If there's like only 1 import in a few categories, feel free to merge things together.